msbuild/Xamarin.MacDev.Tasks.Core/Tasks/PrepareNativeReferencesTaskBase.cs

```diff
@@ -137,9 +137,16 @@ public override bool Execute ()
 				text.AppendLine ();
 			}
 
+			bool skipLinkWithGeneration = false;
 			var linkWith = Path.Combine (IntermediateOutputPath, "LinkWithAttributes.cs");
-			Directory.CreateDirectory (Path.GetDirectoryName (linkWith));
-			File.WriteAllText (linkWith, text.ToString ());
+			if (File.Exists (linkWith)) {
+				string existingLinkWithText = File.ReadAllText (linkWith);
+				skipLinkWithGeneration = String.Equals (existingLinkWithText, text.ToString (), StringComparison.Ordinal);
+			}
+			if (!skipLinkWithGeneration) {
+				Directory.CreateDirectory (Path.GetDirectoryName (linkWith));
+				File.WriteAllText (linkWith, text.ToString ());
```

Comment:
you might be calling `text.ToString()` twice (not sure if it's cached, assuming it's a `StringBuilder`)